platform/darwin/src/MGLMultiPoint.mm

```diff
@@ -103,22 +103,6 @@ - (BOOL)intersectsOverlayBounds:(MGLCoordinateBounds)overlayBounds
     return MGLLatLngBoundsFromCoordinateBounds(_bounds).intersects(MGLLatLngBoundsFromCoordinateBounds(overlayBounds));
 }
 
-- (mbgl::AnnotationSegments)annotationSegments {
-    NSUInteger count = self.pointCount;
-    CLLocationCoordinate2D *coordinates = self.coordinates;
-    
-    mbgl::AnnotationSegment segment;
-    segment.reserve(count);
-    for (NSUInteger i = 0; i < count; i++) {
-        segment.push_back(MGLLatLngFromLocationCoordinate2D(coordinates[i]));
-    }
-    return { segment };
-}
-
-- (mbgl::ShapeAnnotation::Properties)shapeAnnotationPropertiesObjectWithDelegate:(__unused id <MGLMultiPointDelegate>)delegate {
```

Comment:
MGLAnnotation is a public protocol, so we can’t use any C++ types or expect the developer to implement this method themselves for their custom MGLAnnotation classes.

The base model class for things that get turned into `mbgl::ShapeAnnotation`s is currently MGLMultiPoint. There’s no base model class for things that get turned into `mbgl::PointAnnotation`s, because any Objective-C object can conform to the MGLAnnotation protocol and thus be added as a point annotation.

I like the idea of centralizing this shape conversion logic in one place. MGLMultiPoint is a natural place to do this, but the most flexible approach would be to implement a standalone C function similar to `MGLFeaturesFromMBGLFeatures()`.